### ISSUE TITLE
The Future of the Unsplash-App

### ISSUE THREAD

**Issue by newhinton** [nextcloud/unsplash#33]
This issue is here to allow us to discuss the future of this app. Since there are things that changed regarding licensing of images from the unsplash-service, there may be features which may not be possible.

Everyone is invited to join this discussion and help us improve this app!

Currently, one idea is to move away from unsplash as an image-provider, to wikimedia-commons as an example. Suggestions are welcome!

**Comment by jancborchardt** [comment]
I would be happy to change the name. One thing only:

If we want to keep the current users, which is probably a good idea, we would **not** be able to change is the Nextcloud app ID (as far as I know – @MorrisJobke @rullzer correct?) This means it would always be `https://apps.nextcloud.com/apps/unsplash` in the app store, the apps subfolder would stay `unsplash` and on updates Nextcloud would say `New name (unsplash)`. But everything else I think would be changed.

We can simply rename the app in this repository, or we can also move it out of my personal profile into an organization when we find a name. I would prefer that since it’s a group effort anyway.

Then coming to names – does anyone have any ideas? Unsplash is short and nice, but it’s also a specific platform and they don’t use CC0 anymore. As said, Wikimedia Commons would be nice, but just in case we add others as well (like Pixabay which is CC0), maybe it’s best to not use a specific product name.
- Random background
- Nature backgrounds
- Scenery
- Nature

I like the simple "Nature" name. It’s a splash of nature in the otherwise digital interface ("some Nature for Nextcloud"). And yes, I do think we should stick to nature images cause they look nice. :) (And apparently pop psychology says it’s calming.)

What do you think?

**Comment by MorrisJobke** [comment]
> If we want to keep the current users, which is probably a good idea, we would **not** be able to change is the Nextcloud app ID (as far as I know – @MorrisJobke @rullzer correct?)

Correct.

**Comment by newhinton** [comment]
@jancborchard @mwieschollek While im taking a look at wikimedia, i want to make this app more generic, so that we can provide different imageprovider from which an admin can choose. We should consider changing the appname, any suggestions there?

Some silly ideas:
Colourful
Advanced Theming
Random Images

**Comment by newhinton** [comment]
afaik 'nextcloud' is a term which is not allowed in an appname

**Comment by jancborchardt** [comment]
Let’s go for **Splash**. It’s similar to the previous name, versatile and not only limited to nature if that ever changes. Think "splash of color" or "splash screen". :)

**Comment by newhinton** [comment]
@jancborchardt I have started developing a way to serve multiple different providers of images. While we are at it, what provider do we want to serve? currently i'm planing on providing unsplash and wikimedia commons, and if the postprocessing-pr is accepted a provider to serve the default backgroundimage from the normal theming

**Comment by jancborchardt** [comment]
@newhinton I’d say Unsplash and Wikimedia Commons are fine and enough too? There’s also Pixabay and Pexels but they both have the same license as Unsplash so we don’t really need to add them.

**Comment by jancborchardt** [comment]
Heyhey @newhinton @marius-wieschollek – considering I don’t really find the time to work on this app anymore (also as the first prototype already did what I wanted and it’s technically beyond my abilities ;) I’m wondering if you would like to take over maintainership together?

Because otherwise I’d feel like I’d just be blocking progress on the app by being idle, and that’s not nice considering all the good open pull requests for enhancements.

I guess we also have to ask @MorrisJobke how to exactly handle it. If it makes sense I’d like the app to be moved to the https://github.com/nextcloud organization so it also doesn’t just look like I’m the only one working on it.

Let me know what you think! :)

**Comment by newhinton** [comment]
@jancborchardt nice to hear that you are still alive ^^ i don't think this app will require a whole lot of maintenance, so i'd be willing to help out :D

**Comment by MorrisJobke** [comment]
> I guess we also have to ask @MorrisJobke how to exactly handle it. If it makes sense I’d like the app to be moved to the https://github.com/nextcloud organization so it also doesn’t just look like I’m the only one working on it.

Yep - just moving it over should work :)

**Comment by jancborchardt** [comment]
Ok, so then I would move the app over to the Nextcloud organization. :) Just want to wait on an answer from @marius-wieschollek as well since it’s always best to have 2 maintainers at least.

Some other things:
- @newhinton would need to get the (or a new) publishing key for the app store releases
- We adjust the credits in the [readme](https://github.com/jancborchardt/unsplash/blob/master/README.md), and once the Wikimedia Commons info is in also the title
- Basically we rename "Unsplash" to "Splash" everywhere. Except the repo name, cause that would make it incompatible with existing installations.

Anything else? :)

**Comment by marius-wieschollek** [comment]
@jancborchardt I would be willing to help out with the development of the app.

**Comment by newhinton** [comment]
@jancborchardt We would also need to transfer ownership in the nextcloud-appstore if we use new keys

will you have time in the near future that we can discuss it? maybe on irc or something like that?

**Comment by jancborchardt** [comment]
Sorry for the delay!

- [x] moved the app, we are now at https://github.com/nextcloud/unsplash – the repo name needs to stay like that for legacy reasons, but we can rename to "Splash"
- [x] created a team @nextcloud/unsplash for the team, and added @newhinton & @marius-wieschollek as maintainers: https://github.com/orgs/nextcloud/teams/unsplash/members
- [ ] we need to get a new appstore key. @MorrisJobke what’s the process here – I guess @newhinton for example opens a pull request with a new certificate at https://github.com/nextcloud/app-certificate-requests/tree/master/unsplash ?
- [ ] We adjust the credits in the readme, and once the Wikimedia Commons info is in also the title
- [ ] Basically we rename "Unsplash" to "Splash" everywhere. Except the repo name, cause that would make it incompatible with existing installations.
- [ ] anything else?

**Comment by jancborchardt** [comment]
Sorry for the delay!

- [x] moved the app, we are now at https://github.com/nextcloud/unsplash – the repo name needs to stay like that for legacy reasons, but we can rename to "Splash"
- [x] created a team @nextcloud/unsplash for the team, and added @newhinton & @marius-wieschollek as maintainers: https://github.com/orgs/nextcloud/teams/unsplash/members
- [ ] we need to get a new appstore key. @MorrisJobke what’s the process here – I guess @newhinton for example opens a pull request with a new certificate at https://github.com/nextcloud/app-certificate-requests/tree/master/unsplash ?
- [ ] We adjust the credits in the readme, and once the Wikimedia Commons info is in also the title
- [ ] Basically we rename "Unsplash" to "Splash" everywhere. Except the repo name, cause that would make it incompatible with existing installations.
- [ ] anything else?

**Comment by MorrisJobke** [comment]
> [ ] we need to get a new appstore key. @MorrisJobke what’s the process here – I guess @newhinton for example opens a pull request with a new certificate at https://github.com/nextcloud/app-certificate-requests/tree/master/unsplash ?

Why this? Don't you have it anymore? You could reuse that?

**Comment by MorrisJobke** [comment]
And if so: yes new CSR.

**Comment by jancborchardt** [comment]
@MorrisJobke so I just send my part of the cert to @newhinton? I mean, doesn’t it have my ID in it?

**Comment by newhinton** [comment]
@jancborchardt If @MorrisJobke says reusing the same cert is fine, you can contact me unsplash[enterspecialAhere]felixnuesse.de so that we can privately discuss on how to transfer the certs :D

**Comment by MorrisJobke** [comment]
> @MorrisJobke so I just send my part of the cert to @newhinton? I mean, doesn’t it have my ID in it?

Correct. The cert is just a signed proof from our side. You generated the CSR with a generated secret on your side. This allows to sign new app releases and those release signatures can be verified by checking against our root cert, because it signed your secret, that is used to sign the app by yourself. So it is only personalized to you in a way that only you have access to it. So it's fine to give it to the new maintainer and giving the new maintainer this power to sign that app.

**Comment by MorrisJobke** [comment]
Keep in mind that you need to send @newhinton the secret (often called `key`) and not the cert (which is the public part).

**Comment by Hell-o-Admin** [comment]
Hey guys,

since unsplash seems to have had some issues with timeouts yesterday ([https://status.unsplash.com/](url)), which resulted in blank background image on my nextcloud instance, i started reading some issues in this repo.

First of all i wanted to thank you very much for your great work on this app! I love to be suprised by a nice picture on login.

I tought you might be interested in some user feedback and that is why im posting here, even if im not really able to help you (code the app).

Here my to cents for the future of the app and its features:

- I'd really like to see the image source as in @marius-wieschollek solution to issue #9 

- I am not using the feature that replaces the background of the navigation bar, because it often has a random picture with white color or even little contrast to the icons in that area . As a result, you cant see the icons. I wouldnt mind, if you drop this feature.

- If you would change to wikimedia commons or other sources, i'd like to be able to choose the category or multible categories to get the random pictures from, if that would be possible. I think, this is the same as intended in #4 

- Another request which i would back, is to change the randomisation to daily (or weekly). This would maybe also make it possible to cache the picture in the cloud for the day, if the copyrights allow caching at all. E.g. bing.com changes the picture daily and i like that.

Finally thanks for your great work again and if you didnt want any user feedback here, just ignore it. ;-)

Greetings,
Hell-o-Admin

**Comment by newhinton** [comment]
@Hell-o-Admin

thank you for your feedback!

i'll write some lines to your questionv

1) I will need to see if this can added in the future. We have some licensing issues which block his pr until we take a closer look at it. However, steps have been taken to include this in the future (but don't wait for it, it may take a while)

2) You should already be able to disable this on a per user basis, but there will be the ability to further modify this in the near future

3) I have already taken steps to implement that, take a look at#44

However, the api is not easy to understand and currently only dogs (and cats) are returned as images. Help here would be greatly welcomed :)

4) This was already implemented by @marius-wieschollek in his pr regarding the image attribution, so we need to look at this at well.

**Comment by jancborchardt** [comment]
I guess we can close this issue as it’s transferred? :)

@hell-o-admin thanks or the feedback! It’s always best to check if issues already exist (in that case comment there) or if they don’t open separate issues for separate things.
